daphne/src/vdaf/mod.rs

```diff
@@ -196,17 +195,18 @@ impl VdafConfig {
         extensions: &[u8],
         encrypted_input_share: &HpkeCiphertext,
     ) -> Result<(VdafState, VdafMessage), DapError> {
-        let mut info = Vec::with_capacity(CTX_INPUT_SHARE_PREFIX.len() + 34);
-        task_id.encode(&mut info);
-        info.extend_from_slice(CTX_INPUT_SHARE_PREFIX);
-        info.push(CTX_ROLE_CLIENT);
-        info.push(if is_leader {
+        const N: usize = CTX_INPUT_SHARE.len();
+        let mut info = [0; N + 2];
+        info[..N].copy_from_slice(CTX_INPUT_SHARE);
+        info[N] = CTX_ROLE_CLIENT; // Sender role
+        info[N + 1] = if is_leader {
             CTX_ROLE_LEADER
         } else {
             CTX_ROLE_HELPER
-        });
+        }; // Redeiver role
 
-        let mut aad = Vec::with_capacity(10);
+        let mut aad = Vec::with_capacity(50 + extensions.len());
+        task_id.encode(&mut aad);
         nonce.encode(&mut aad);
```

Comment:
is it ok the calculation of the nonce length? now task_id was prepended.

Reply:
Nope, it's wrong!! I discovered that when confirming interop with Janus. Will fix in this PR before merging.

Reply:
Fixed. Great catch!